web-common/src/features/entity-management/resources-store.ts

```diff
@@ -0,0 +1,128 @@
+import {
+  ResourceKind,
+  useResource,
+} from "@rilldata/web-common/features/entity-management/resource-selectors";
+import type {
+  V1ParseError,
+  V1Resource,
+  V1ResourceName,
+} from "@rilldata/web-common/runtime-client";
+import type { CreateQueryResult } from "@tanstack/svelte-query";
+import { derived, get, Readable, writable } from "svelte/store";
+
+/**
+ * Global resources store that maps file name to a resource and parse errors.
+ */
+export type ResourcesState = {
+  // this is just a mapping of file path to resource name
+  // storing the entire resource is not necessary since tanstack query will do that for the get resource api
+  resources: Record<string, V1ResourceName>;
+  errors: Record<string, Array<V1ParseError>>;
```

Comment:
I’m thinking that tracking a file’s parse errors in this store is overloading the store’s purpose. A file with a parse error isn't yet associated with a resource.

Would it make sense to instead have a `useParseErrors(instanceId: string, fileName: string)` selector that operates on top of a TanStack Query call to the `GetResource` API for the `ProjectParser` resource?

Reply:
This is more performant. Instead of filtering through project parse errors for each instance of get errors we have a map at a single place. Lets think about performance from the get go.

Reply:
It also allows for getting errors from different files that map to a single resource easily.

Reply:
Yeah, I see the performance optimization. But we're filtering a list of files/errors – wouldn't a project have to be crazy big for this to be human-perceptible?

My concern was with the semantics – a file with a parse error isn't associated with a resource (source/model/metrics view), so it feels weird that it'd be in a `resources-store`. If keeping this approach, maybe just a better name would be the `files-store`?